acceptance/tests/test_upgrade_puppet6_to_puppet7.rb

```diff
@@ -0,0 +1,52 @@
+require 'beaker-puppet'
+require_relative '../helpers'
+
+# Tests FOSS upgrades from the latest puppet 6 (the puppet6 collection) to puppet 7.0.0.
+test_name 'puppet_agent class: Upgrade agents from puppet6 to puppet7' do
+  source = ""
+  if agents_only.any? { |agent| /^fedora-|^el-/ =~ agent.platform }
+    source = "yum_source => 'http://nightlies.puppet.com/yum'"
+  elsif  agents_only.any? { |agent| /^debian-|^ubuntu-/ =~ agent.platform }
+    source = "apt_source => 'http://nightlies.puppet.com/apt'"
+  else
+    skip_test "No suitable agents present"
+  end
+
+  # require_master_collection 'puppet7' #once puppet7 collection is available this can be uncommented
+  exclude_pe_upgrade_platforms
+  latest_version = `curl http://builds.delivery.puppetlabs.net/passing-agent-SHAs/puppet-agent-main-version`.gsub(/\d*\.\d*.\d*\./,"7.0.0.")
+  
+  puppet_testing_environment = new_puppet_testing_environment
+  step "Create new site.pp with upgrade manifest" do
+    manifest = <<-PP
+    node default {
+      class { puppet_agent:
+        package_version => '#{latest_version}',
+        #{source},
```

Comment:
Actually, I think you can have both `apt_source` and `yum_source` in the same class and each OS will use what it needs.

Reply:
Yep, just tested and it worked :))